tests/test_dcm/test_header_tweaks.py

```diff
@@ -0,0 +1,208 @@
+                    {
+                        'ROIPhysicalProperty': 'EFFECTIVE_Z',
+                        'ROIPhysicalPropertyValue': 6
+                    }
+                ]
+            },
+            {
+                'ReferencedROINumber': 2,
+            },
+            {
+                'ReferencedROINumber': 10,
+            },
+            {
+                'ReferencedROINumber': 99,
+                'ROIPhysicalPropertiesSequence': [
+                    {
+                        'ROIPhysicalProperty': 'REL_ELEC_DENSITY',
+                        'ROIPhysicalPropertyValue': 0
+                    }
+                ]
+            }
+        ]
+    })
+
+    expected_dicom_file = dcm_from_dict({
+        'RTROIObservationsSequence': [
+            {
+                'ReferencedROINumber': 1,
+                'ROIPhysicalPropertiesSequence': [
+                    {
+                        'ROIPhysicalProperty': 'EFFECTIVE_Z',
+                        'ROIPhysicalPropertyValue': 6
+                    },
+                    {
+                        'ROIPhysicalProperty': 'REL_ELEC_DENSITY',
+                        'ROIPhysicalPropertyValue': adjustment_map['to_be_changed 1']
+                    }
+                ]
+            },
+            {
+                'ReferencedROINumber': 2
+            },
+            {
+                'ReferencedROINumber': 10,
+                'ROIPhysicalPropertiesSequence': [
+                    {
+                        'ROIPhysicalProperty': 'REL_ELEC_DENSITY',
+                        'ROIPhysicalPropertyValue': adjustment_map['to_be_changed 2']
+                    }
+                ]
+            },
+            {
+                'ReferencedROINumber': 99,
+                'ROIPhysicalPropertiesSequence': [
+                    {
+                        'ROIPhysicalProperty': 'REL_ELEC_DENSITY',
+                        'ROIPhysicalPropertyValue': adjustment_map['to_be_changed 3']
+                    }
+                ]
+            }
+        ]
+    }, template_dcm=original_dicom_file)
+
+    adjusted_dicom_file = adjust_rel_elec_density(
+        original_dicom_file, adjustment_map)
+
+    assert adjusted_dicom_file != original_dicom_file
+    assert str(expected_dicom_file) == str(adjusted_dicom_file)
+
+    pydicom.write_file(ORIGINAL_DICOM_FILENAME, original_dicom_file)
+    adjustment_map_as_list = [
+        ['{}'.format(key), item] for key, item in adjustment_map.items()
+    ]
+    adjustment_map_flat = np.concatenate(adjustment_map_as_list).tolist()
+
+    command = (
+        'pymedphys dicom adjust-rel-elec-density'.split()
+        + [ORIGINAL_DICOM_FILENAME, ADJUSTED_DICOM_FILENAME]
+        + adjustment_map_flat)
+
+    print(command)
+
+    subprocess.check_call(command)
+
+    cli_adjusted_dicom_file = pydicom.read_file(
+        ADJUSTED_DICOM_FILENAME, force=True)
+
+    assert str(expected_dicom_file) == str(cli_adjusted_dicom_file)
```

Comment:
By using `str` here pytest produces a really nice diff print out displaying where the DICOM files are different.